### DISCUSSION TITLE
OPA for Windows Checksums

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#377] [OPA and Rego] [answered]
Hi There,

I'm currently reading https://www.openpolicyagent.org/docs/latest/#1-download-opa. I have a question about OPA for Windows checksums. The file does not seem to exist, https://openpolicyagent.org/downloads/v0.50.2/opa_windows_amd64.sha256.
Could you please clarify?

Thanks!

**Comment by srenatus** [comment, marked as answer]
It's .sha256, not .shar256. You'll find the file here: https://github.com/open-policy-agent/opa/releases/download/v0.50.2/opa_windows_amd64.exe.sha256

**Reply by Ronnie-personal** [reply]
Thanks for the info. It works!
Shall we update the document?

Thanks!

**Reply by srenatus** [reply]
The original URL is correct, too:
```
$ curl https://openpolicyagent.org/downloads/v0.50.2/opa_windows_amd64.sha256 
Redirecting to https://www.openpolicyagent.org/downloads/v0.50.2/opa_windows_amd64.sha256    
```

the problem was *shar* instead of *sha* in your URL. If that's in the docs somewhere, it should be corrected.

**Reply by Ronnie-personal** [reply]
I tested using sha256, I didn't use anything like shar256.
When open the link from browser, it shows for 404.

**Reply by srenatus** [reply]
Must have misread something, sorry. I'll have another look later.

**Reply by srenatus** [reply]
Sorry. Indeed, that's the wrong URL. We're directing OK, but the redirect target is wrong.

Ah I see what's happening. We're missing the `.exe`:  https://openpolicyagent.org/downloads/v0.50.2/opa_windows_amd64.exe.sha256 👈 This one works.

**Reply by Ronnie-personal** [reply]
Thank you so much for helping on the question.

Now I understand why I don't have issue to download Linux checksum, but only have issue to download windows checksum file.

 'sha265' should be appended to windows binary file name, including the extension 'exe'.

I thought only the file name, not the file extension.

**Reply by srenatus** [reply]
The linux binaries have no extension, so it's `linux_amd64_static` -> `linux_amd64_static.sha256`. I suppose adding that information to the windows info box, 
> Windows users can obtain the OPA executable from [here](https://openpolicyagent.org/downloads/v0.50.2/opa_windows_amd64.exe). The steps below are the same for Windows users except the executable name will be different.

Would be a good addition. Would you be up to contributing a PR with that?

**Reply by Ronnie-personal** [reply]
Sure, I am happy to update the document.

**Reply by Ronnie-personal** [reply]
@srenatus  I have created PR https://github.com/open-policy-agent/opa/pull/5803/files

**Reply by srenatus** [reply]
Thanks!
